drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -277,43 +286,63 @@ static void _receive_data(at86rf2xx_t *dev)
     if (dev->options & AT86RF2XX_OPT_RAWDUMP) {
         payload = gnrc_pktbuf_add(NULL, NULL, pkt_len, GNRC_NETTYPE_UNDEF);
         if (payload == NULL ) {
+            at86rf2xx_fb_stop(dev);
             DEBUG("[at86rf2xx] error: unable to allocate RAW data\n");
             return;
         }
-        at86rf2xx_rx_read(dev, payload->data, pkt_len, 0);
+        at86rf2xx_fb_read(dev, payload->data, pkt_len);
+        at86rf2xx_fb_stop(dev);
         dev->event_cb(NETDEV_EVENT_RX_COMPLETE, payload);
         return;
     }
 
     /* get FCF field and compute 802.15.4 header length */
-    at86rf2xx_rx_read(dev, mhr, 2, 0);
+    at86rf2xx_fb_read(dev, mhr, 2);
+
     hdr_len = _get_frame_hdr_len(mhr);
     if (hdr_len == 0) {
+        at86rf2xx_fb_stop(dev);
         DEBUG("[at86rf2xx] error: unable parse incoming frame header\n");
         return;
     }
+
     /* read the rest of the header and parse the netif header from it */
-    at86rf2xx_rx_read(dev, &(mhr[2]), hdr_len - 2, 2);
+    at86rf2xx_fb_read(dev, &(mhr[2]), hdr_len - 2);
     hdr = _make_netif_hdr(mhr);
     if (hdr == NULL) {
+        at86rf2xx_fb_stop(dev);
         DEBUG("[at86rf2xx] error: unable to allocate netif header\n");
         return;
     }
+
     /* fill missing fields in netif header */
     netif = (gnrc_netif_hdr_t *)hdr->data;
     netif->if_pid = dev->mac_pid;
-    at86rf2xx_rx_read(dev, &(netif->lqi), 1, pkt_len);
-    netif->rssi = at86rf2xx_reg_read(dev, AT86RF2XX_REG__PHY_ED_LEVEL);
 
     /* allocate payload */
     payload = gnrc_pktbuf_add(hdr, NULL, (pkt_len - hdr_len), dev->proto);
     if (payload == NULL) {
+        at86rf2xx_fb_stop(dev);
         DEBUG("[at86rf2xx] error: unable to allocate incoming payload\n");
         gnrc_pktbuf_release(hdr);
         return;
     }
     /* copy payload */
-    at86rf2xx_rx_read(dev, payload->data, payload->size, hdr_len);
+    at86rf2xx_fb_read(dev, payload->data, payload->size);
+
+    /* Ignore FCS but advance fb read */
+    at86rf2xx_fb_read(dev, NULL, 2);
+
+    at86rf2xx_fb_read(dev, &(netif->lqi), 1);
+
+#ifndef MODULE_AT86RF231
```

Comment:
Now I have :smile: Yes, it has ED value after LQI, so it's fine. See [p.133](http://www.atmel.com/Images/Atmel-42002-MCU_Wireless-AT86RF212B_Datasheet.pdf)